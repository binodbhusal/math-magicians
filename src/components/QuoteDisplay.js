import React, { useEffect, useState } from 'react';

const DisplayQoute = () => {
  const [qoute, setqoute] = useState('');
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  const quoteApi = 'https://api.api-ninjas.com/v1/quotes?category=';
  const token = 'NpIyy7AoiV/za/U0ht20ew==4F4oVlmUzGUvmLMN';

  const fetchApi = async () => {
    try {
      const response = await fetch(quoteApi, {
        headers: {
          'X-Api-Key': token,
          'content-type': 'application/json',
        },
      });
      const data = await response.json();
      setqoute(data[0].quote);
      setLoading(false);
    } catch (error) {
      setError(error);
      setLoading(false);
    }
  };
  useEffect(() => {
    fetchApi();
  }, []);
  if (loading) {
    return <p>Loading...</p>;
  }
  if (error) {
    return (
      <p>
        Error:
        {' '}
        {error.message}
      </p>
    );
  }
  return (
    <div className="qoute-container">
      <h2>Qoute of the Day</h2>
      <p>{qoute}</p>
    </div>
  );
};

export default DisplayQoute;
