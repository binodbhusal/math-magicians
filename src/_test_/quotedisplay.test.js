import React from 'react';
import { render } from '@testing-library/react';
import QuoteDisplay from '../components/QuoteDisplay';

describe('Tests Quote component ', () => {
  it('renders correctly', () => {
    const tree = render(<QuoteDisplay />);
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    render(<QuoteDisplay />);
  });
});
