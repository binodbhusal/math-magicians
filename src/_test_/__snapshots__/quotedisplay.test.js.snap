// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders quote display correctly 1`] = `
<p>
  Loading...
</p>
`;
